cluster/manifests/skipper/deployment.yaml

```diff
@@ -1,29 +1,48 @@
+{{- if eq .ConfigItems.skipper_canary_enabled "true" }}
```

Comment:
This does not look right because it conditions whole manifest on  skipper_canary_enabled.
I think we need something like:
```
{{ if eq .ConfigItems.skipper_canary_enabled "true" }}
{{ $deployment_config := `[two items: main and canary]` }}
{{ else }}
{{ $deployment_config := `[one item: main]` }}
{{ end }}

{{ range json $deployment_config }}
...
```

Reply:
I think there might be variable scope issue with the above though.

Reply:
I think it would need declaration and assignments like so 
```
{{ $deployment_config := "" }}
{{ if eq .ConfigItems.skipper_canary_enabled "true" }}
  {{ $deployment_config = `[two items: main and canary]` }}
{{ else }}
  {{ $deployment_config = `[one item: main]` }}
{{ end }}

{{ range json $deployment_config }}
...
```
https://stackoverflow.com/questions/68814385/how-do-i-conditionally-declare-variables-in-go-templates

Reply:
What about the 2nd commit?
Is this possible?